crypto/stack/stack.c

```diff
@@ -191,7 +191,7 @@ static int internal_find(OPENSSL_STACK *st, const void *data,
                         ret_val_options);
     if (r == NULL)
         return (-1);
-    return (int)((char **)r - st->data);
+    return (int)((const char **)r - st->data);
```

Comment:
`r` is declared as a `const void * const *` but `OBJ_bsearch_ex_` returns a `const void *` and the only interesting usage of `r` casts it to `const char **`. Surely we should either declare it to be the same as the return type of `OBJ_bsearch_exe_` or the same as the type of where we use it.